### PR TITLE
Fix header limit exceeded

### PR DESCRIPTION
**Pull request by s-matyukevich**
This PR fixes https://github.com/grpc/grpc-go/issues/4265 by returning a small hardcoded error message to the client in cases when header size limit is exceeded, instead of closing the connection without passing any error. This implements the proposed solution from the issue description.

In the https://github.com/grpc/grpc-go/issues/4265 @easwars also suggested a different approach: ignore header size limit on the server and let the client handle it. Please let me know if I should implement this suggestion instead.

### CODE REVIEW

File: internal/transport/http2_server.go [hunk 1]
```diff
@@ -1064,6 +1075,22 @@ func (t *http2Server) WriteStatus(s *Stream, st *status.Status) error {
 	return nil
 }
 
+// replaceStatusInHeaders replaces "grpc-status", "grpc-message" and "grpc-status-details-bin" fields.
+// This is useful in cases when error message is too large and can't fit into the liit set by the client.
```
Comment (evanj) [inline internal/transport/http2_server.go:1079]: I think there is a typo: liit -> limit?

**Review by evanj — COMMENTED** [review]
This would be a substantial improvement over the current situation, thanks! Now we need to do the same thing for the C and Java servers ...

**Comment by arvindbr8** [comment]
HI @s-matyukevich,

thanks for the PR. 

Per discussions in #4265 it seemed like we were more inclined toward ignoring header size limit on the server and let the client handle it. There are more comments above how we want to handle it as well and the edge cases. 

Please let know in the issue there if you have more questions.

File: internal/transport/http2_server.go (hunk 1 above)
Comment (s-matyukevich) [reply internal/transport/http2_server.go]: fixed.

**Comment by s-matyukevich** [comment]
@arvindbr8 

Ok, I'll implement the logic to ignore header size on the server instead.

**Comment by arvindbr8** [comment]
Will continue the discussion in the issue #4265. We can create a new PR once the decisions are made.

**Pull request closed without merging**